core/src/main/java/com/linecorp/armeria/common/thrift/ThriftCallbackListenableFuture.java

```diff
@@ -0,0 +1,23 @@
+package com.linecorp.armeria.common.thrift;
+
+import org.apache.thrift.async.AsyncMethodCallback;
+
+import com.google.common.util.concurrent.AbstractFuture;
+import com.google.common.util.concurrent.ListenableFuture;
+
+/**
+ * A {@link ListenableFuture} that can be passed in as an {@link AsyncMethodCallback}
+ * when making an asynchronous thrift client rpc.
+ */
+public class ThriftCallbackListenableFuture<T> extends AbstractFuture<T> implements AsyncMethodCallback<T> {
+
```

Comment:
How about adding two more constructors (successful and failed) as well as the default one for easier construction?